cmd/oceantv/broadcast_hardware_machine.go

```diff
@@ -247,6 +259,20 @@ func (sm *hardwareStateMachine) handleHardwareResetRequestEvent(event hardwareRe
 	}
 }
 
+func (sm *hardwareStateMachine) handleControllerFailureEvent(event controllerFailureEvent) {
+	sm.log("handling controller failure event")
+	switch sm.currentState.(type) {
+	case *hardwareOn, *hardwareRestarting, *hardwareStarting:
+		sm.transition(newHardwareStopping(sm.ctx))
```

Comment:
I'm almost wondering if this should be a transition straight to hardwareOff ? If there's an ESP failure then are we really stopping anything; isn't it already stopped ?

Reply:
Done.